internal/provider/kubernetes/topology_injector.go

```diff
@@ -49,10 +50,28 @@ func (m *ProxyTopologyInjector) Handle(ctx context.Context, req admission.Reques
 	}
 
 	pod := &corev1.Pod{}
-	if err := m.Get(ctx, podName, pod); err != nil {
-		logger.Error(err, "get pod failed", "pod", podName.String())
-		topologyInjectorEventsTotal.WithFailure(metrics.ReasonError).Increment()
-		return admission.Allowed("internal error, skipped")
+	// Cache isn't guaranteed to be updated yet, so we retry a few times.
+	var getErr error
+	for i := 0; i < 2; i++ {
```

Comment:
this has an impact/delay on pod creation for all pods, so prefer if we removed the retry and relied on the APIReader

Reply:
It's a 10ms wait each loop with a max of 20ms. Wouldn't this be less taxing vs opening a connection to the API server?

How about reducing to 5ms and/or only a single retry?

Reply:
the APIReader is the last resort, so that should def go in, the retry is an optimization but also has some downside so prefer if we added it later after gathering more data on how it impacts large systems